include/dlaf/permutations/general.h

```diff
@@ -36,25 +36,28 @@ namespace dlaf::permutations {
 template <Backend B, Device D, class T, Coord coord>
 void permute(SizeType i_begin, SizeType i_end, Matrix<const SizeType, D>& perms,
              Matrix<const T, D>& mat_in, Matrix<T, D>& mat_out) {
-  const matrix::Distribution& distr_perms = perms.distribution();
-  const matrix::Distribution& distr_in = mat_in.distribution();
-  const matrix::Distribution& distr_out = mat_out.distribution();
-
   DLAF_ASSERT(matrix::local_matrix(perms), perms);
   DLAF_ASSERT(matrix::local_matrix(mat_in), mat_in);
   DLAF_ASSERT(matrix::local_matrix(mat_out), mat_out);
 
+  // Note:
+  // These are not implementation constraints, but more logic constraints. Indeed, these ensure that
+  // the range [i_begin, i_end] is square in terms of elements (it would not make sense to have it square
```

Comment:
`i_end` is not included in the range anymore
```suggestion
  // the range [i_begin, i_end) is square in terms of elements (it would not make sense to have it square
```